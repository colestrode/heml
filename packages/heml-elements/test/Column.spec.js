const render = require('@heml/render')
const Column = require('../build/Column').default
const Style = require('../build/Style').default

beforeAll(() => {
  Style.preRender({ elements: [] });
})

test('renders with not attrs or content', () => {
  return expect(render.renderElement(Column, {} )).toMatchSnapshot()
});

test('renders with content', () => {
  return expect(render.renderElement(Column, {}, 'Lady Dynamite')).toMatchSnapshot()
})

test('renders with small attr', () => {
  const attrs = {
    small: 6
  }

  return expect(render.renderElement(Column, attrs)).toMatchSnapshot()
})

test('renders with large attr', () => {
  const attrs = {
    large: 12
  }

  return expect(render.renderElement(Column, attrs)).toMatchSnapshot()
})

test('renders with large the same as small', () => {
  const attrs = {
    large: 12,
    small: 12
  }

  return expect(render.renderElement(Column, attrs)).toMatchSnapshot()
})
