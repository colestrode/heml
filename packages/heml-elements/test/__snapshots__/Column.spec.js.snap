// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with content 1`] = `"<td class=\\"column col-sm-12\\" width=\\"100%\\" style=\\"width: 100%;\\" align=\\"left\\" valign=\\"top\\">Lady Dynamite</td>"`;

exports[`renders with large attr 1`] = `"<td class=\\"column col-sm-12\\" width=\\"100%\\" style=\\"width: 100%;\\" align=\\"left\\" valign=\\"top\\">&nbsp;</td>"`;

exports[`renders with large the same as small 1`] = `"<td class=\\"column col-sm-12\\" width=\\"100%\\" style=\\"width: 100%;\\" align=\\"left\\" valign=\\"top\\">&nbsp;</td>"`;

exports[`renders with not attrs or content 1`] = `"<td class=\\"column col-sm-12\\" width=\\"100%\\" style=\\"width: 100%;\\" align=\\"left\\" valign=\\"top\\">&nbsp;</td>"`;

exports[`renders with small attr 1`] = `"<td class=\\"column col-sm-6\\" width=\\"100%\\" style=\\"width: 100%;\\" align=\\"left\\" valign=\\"top\\">&nbsp;</td>"`;
